### PR TITLE
Fixes top-level fields in arm.Resource marshaling

### PR DESCRIPTION
**Pull request by m1kola**
Also simplifies arm.Resource#MarshalJSON: no need to use orderedmap here.

### CODE REVIEW

File: pkg/util/arm/arm_test.go [hunk 1]
```diff
@@ -115,6 +122,10 @@ type testResource struct {
 	Name        string                   `json:"name,omitempty"`
 	Unmarshaled int                      `json:"-"`
 	unexported  int
+	// Both `arm.Resource` and nested `testResource` have fields with name `Tags`.
+	// The `Tags` field from `arm.Resource` must override the one from `testResource`
+	// on the top-level of JSON.
+	Tags map[string]*string `json:"tags"`
```
Comment (m1kola) [inline pkg/util/arm/arm_test.go:128]: In the real Azure SDK structs almost all resources have similar `Tags` field.

Our custom `arm.Resource` also has the `Tags` field defined like this:

```go
Tags       map[string]interface{} `json:"tags,omitempty"`
```

Despite the fact that outer`Tags` field has `omitempty`, on a structure like this:

```go
&arm.Resource{
    Name:     "test",
    Resource: &testResource{},
}
```

we will get a json like this:

```json
{
  "name": "test",
  "tags": null
}
```

I believe `tags` should be omited in this case.

Comment (m1kola) [reply pkg/util/arm/arm_test.go:128]: **NOTE:** `"tags": null` was one of the reasons why vmss hashes were different in v3. I updated the implementation in https://github.com/openshift/openshift-azure/pull/2101 for consistency as well.

File: pkg/util/arm/arm_test.go [hunk 2]
```diff
@@ -47,28 +47,35 @@ func TestResourceMarshal(t *testing.T) {
...
+      "float": 1.1,
+      "tags": null
     }
   ],
   "byte_slice": "dGVzdA==",
   "string": "test",
   "struct": {
-    "string": "test"
+    "string": "test",
+    "tags": null
```
Comment (m1kola) [inline pkg/util/arm/arm_test.go]: Note that `"tags": null` is omited on the top-level (the `Tags` field was overiden by the one from `arm.Resource`), but not on the deeper levels.

Othervise - order of the fields is the same.

**Comment by m1kola** [comment]
Rebased on top of `master`.

**Comment by jim-minter** [comment]
missing a `go generate ./...` - caught by github actions :)
have pushed an additional commit to your branch

**Pull request merged**